### ISSUE TITLE
Dev docs, build a package - step 7 issues due to manifest

### ISSUE THREAD

**Issue by spencerflagg** [Start9Labs/documentation#104]
https://start9.com/latest/developer-docs/build-package-example/07_create-s9pk

***TL;DR: I don't think this documentation is up-to-date with the hello-world project***

### Errors when running `embassy-sdk pack`

1) The manifest code from [step 5](https://start9.com/latest/developer-docs/build-package-example/05_create-manifest.html) is not identical to the manifest.yaml from the hello-world project, and has indenting issues within the `Description` struct, causing
```
Deserialization Error: invalid type: null, expected struct Description
```

2) After fixing this, still getting the following 
```
Filesystem I/O Error: image.tar: No such file or directory (os error 2)
```
probably because I was not instructed to create a `Makefile` in the previous steps?

**Comment by kn0wmad** [comment]
I believe `min-os-version` needs to be removed - can you test?

**Comment by spencerflagg** [comment]
When I got the I/O error, I was already using the manifest.yaml that I'd copied from the Hello World Wrapper project, and in that version there's no `min-os-version`.

**Comment by ProofOfKeags** [comment]
yeah min-os-version has been deprecated and should be removed from the docs.

**Comment by ProofOfKeags** [comment]
> After fixing this, still getting the following
Filesystem I/O Error: image.tar: No such file or directory (os error 2)
probably because I was not instructed to create a Makefile in the previous steps?

fwiw Makefiles are definitely not required. They are a convention we use internally, but you can use whatever build system you feel comfortable with. The part that we do enforce to make sure that the tools all work is really what is required by the `embassy-sdk pack` step which requires you to have a docker image named with the same name as specified in the manifest.yaml's `.assets.docker-images`

I suspect that the problem you are encountering is that you aren't creating an image.tar (which is the default name if unspecified) and so the pack step is looking for a docker image that doesn't exist.

Of course, all of this should be made clear in the docs and clearly we lost you somewhere, so there is something to be done here. However, at the moment I'm not quite sure how best to make this more clear. Let's start by getting you unstuck and if that resolves it, we can figure out a way to get that information into the docs better.

**Comment by spencerflagg** [comment]
Roger.

There's nothing in steps 1-7 that **explicitly** describes creating an image.tar - I assume that it gets created as the result of some other step?

So far, I haven't spent much time diagnosing the issues - I've only been attempting to follow along in the [/build-package-example/](https://start9.com/latest/developer-docs/build-package-example/) docs and documenting where I had trouble. There's been other issues with those docs that have been discussed in the dev chat, and I already submitted a fix to step 5. I don't think anyone has actually gotten through all the steps without issue.

If it's more helpful, I can try following the [hello-world-wrapper](https://github.com/Start9Labs/hello-world-wrapper) docs instead and once that builds, do a diff on the working directories and see what the issue is.

**Comment by ProofOfKeags** [comment]
Yeah I think the actionable step here is describing the requirements of the embassy-sdk pack step and saying that it looks for the docker image located in the file specified in `.assets.docker-images` AND specify that if it is omitted it will look for `image.tar`. From there we can also include an example `docker build ...` command that shows what it takes to actually *create* said file. This will give the information needed without overspecifying the build system you have to use.

**Comment by gStart9** [comment]
Some of this was updated by #348

After these changes I had no trouble building hello world (on Debian).

I believe the image.tar issues were addressed with #192 and possibly previous PRs.

Closing this for now if no one objects.  Still, these dev docs will need a thorough review by a dev (@elvece @Blu-J @k0gen @islandbitcoin), especially after the StartSDK release and v0.3.5.

**Comment by kn0wmad** [comment]
> Some of this was updated by #348
> 
> After these changes I had no trouble building hello world (on Debian).
> 
> I believe the image.tar issues were addressed with #192 and possibly previous PRs.
> 
> Closing this for now if no one objects. Still, these dev docs will need a thorough review by a dev (@elvece @Blu-J @k0gen @islandbitcoin), especially after the StartSDK release and v0.3.5.

Thanks G - an entire docs rework is scheduled for v040
